crates/hir-ty/src/method_resolution.rs

```diff
@@ -210,9 +210,21 @@ impl TraitImpls {
                 map.entry(target_trait).or_default().entry(self_ty_fp).or_default().push(impl_id);
             }
 
-            // To better support custom derives, collect impls in all unnamed const items.
+            // To better support custom derives, collect impls in all named and unnamed const items.
+            // const FOO: () = { ... };
             // const _: () = { ... };
-            for konst in module_data.scope.unnamed_consts() {
+            for konst in module_data
+                .scope
+                .declarations()
+                .filter_map(|def| {
+                    if let hir_def::ModuleDefId::ConstId(konst) = def {
+                        Some(konst)
+                    } else {
+                        None
+                    }
+                })
+                .chain(module_data.scope.unnamed_consts())
+            {
```

Comment:
This is intentional, `_` is special cased as a macro pattern to support here. Considering all consts for this is too expensive hence why this is done. Future rust will also lint against this "impl in non-`_` const" pattern

Reply:
Oh, thanks. Then I'll try in another way